quarkc/test/lib/conftest.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+import json, sys
+import pexpect
+
+def pytest_collect_file(path, parent):
+    if path.ext == ".q" and path.basename.endswith("_test.q"):
+        return QuarkFile(path, parent)
+
+class QuarkFile(pytest.File):
+    def collect(self):
+        for lang in ["python", "java", "javascript", "ruby"]:
+            yield QuarkItem(self, lang)
+
+class QuarkItem(pytest.Item):
+    def __init__(self, parent, lang):
+        super(QuarkItem, self).__init__("%s:%s" % (lang, parent.fspath.basename), parent)
+        self.lang = lang
+
+    def runtest(self):
+        child = pexpect.spawn("quark", ["install", "-v"] +
+                              ["--%s" % self.lang,
+                               self.parent.fspath.strpath])
+        child.logfile = sys.stdout
+        child.expect("Done", timeout=300)
+        child.expect(pexpect.EOF)
```

Comment:
Expecting `Done` means that every compile error yields a pointless 300 second wait. Just expect `EOF` and then check the result. From test_env_tracing:

        child = pexpect.spawn("quark install --%s %s" % (" --".join(languages), quark_file_name), logfile=sys.stdout)
        child.expect(pexpect.EOF, timeout=300)
        assert child.before.splitlines()[-1].strip() == "Done"